### PR TITLE
[SILOptimizer] Handle a missing case in DataflowDiagnostics.cpp

### PR DESCRIPTION
**Pull request by graydon**
Tiny fix for rdar://34111449

### CODE REVIEW

**Comment by graydon** [comment]
@swift-ci please test

File: test/SILOptimizer/return.swift [hunk 1]
```diff
@@ -94,6 +94,12 @@ func whileLoop(flag: Bool) -> Int {
   }
 } //expected-error {{missing return in a function expected to return 'Int'}}
 
+struct S {}
+extension S:ExpressibleByStringLiteral {
+  init!(stringLiteral:String) {
+  } // expected-error {{missing return in a function expected to return 'S!'}}
```
Comment (slavapestov) [inline test/SILOptimizer/return.swift]: The real error is that we're missing a self.init delegation or self assignment, but I guess before we weren't diagnosing anything at all here?

Comment (graydon) [reply test/SILOptimizer/return.swift]: We were crashing with `missing case` before.

Comment (slavapestov) [reply test/SILOptimizer/return.swift]: I see. LGTM!

**Comment by gottesmm** [comment]
@graydon Next time can you add more context to the commit message on what the crash was? If I didn't have access to radar and I didn't see your comment, I would be very confuzzled = (.

**Comment by graydon** [comment]
@gottesmm oh yeah, good point, sorry!

**Comment by gottesmm** [comment]
@graydon No worries!

**Pull request merged**